### ISSUE TITLE
Maintenance status

### ISSUE THREAD

**Issue by jcgruenhage** [polarlabs/osquery-rust#2]
I've started looking into writing osquery extensions in Rust, and stumbled over this project. It looks quite unfinished right now, and I was wondering what the state here is, considering that there hasn't been any work on it since early this year. Is this something you're still interested in, if there was some more support from other people?

**Comment by jcgruenhage** [comment]
@tokcum hey there, it's been a month. I'm considering forking, but I'd prefer to collaborate upstream instead.

**Comment by jasonk000** [comment]
Hi @jcgruenhage , agree, this looks like an interesting project. Did you take it forward anywhere?

I see the thrift PR has been merged upstream. For now I am using the osquery-go binding but I would much prefer to use rust. I'm not really in a place to build/maintain a complete rust implementation though.

**Comment by jcgruenhage** [comment]
I didn't take it any further, no.

**Comment by tokcum** [comment]
Thank you for your interest in this library. I really appreciate it and I'm sorry, I got not aware of this issue earlier.

True, it has been a while since I worked on the rust bindings for osquery. My original plan was to implement an osquery plugin in rust. I can't exactly remember what this plugin was about. I think it was something I missed at work.

Then I realized that there were no rust bindings. And then I realized that I had to patch Apache Thrift first. I remember that working on the Apache Thrift project was fun, however getting the PR merged and released took ages! It was frustrating. So my priorities shifted to other dev projects.

So, I think the next step for me would be to update osquery-rust and publish it in a proper state based on the updated Thrift crate.

I'm looking forward to support your own projects with a useful rust binding crate. It would be great if we could collaborate to find a good API design for the bindings.

Let me know what you think.
